File: src/bin/0206-reverse-linked-list.cpp
// @leetup=info id=206 lang=cpp slug=reverse-linked-list

#include <algorithm>
#include <iostream>
#include <map>
#include <set>
#include <tuple>
#include <vector>

// Definition for singly-linked list.
struct ListNode {
  int val;
  ListNode* next;

  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode* next) : val(x), next(next) {}
};
// @leetup=code

class Solution {
 public:
  ListNode* reverseList(ListNode* head) {
    ListNode* prev = nullptr;
    ListNode* current = head;
    while (current) {
      ListNode* next_node = current->next;
      current->next = prev;
      prev = current;
      current = next_node;

    }
    return prev;
  }
};
// @leetup=code

int main() {
  std::cout << "Hello, world!" << std::endl;
  return 0;
}

File: src/bin/0206-reverse-linked-list.py
# @leetup=info id=206 lang=python3 slug=reverse-linked-list




class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


# @leetup=code


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        current = head
        while current:
            next_node = current.next
            current.next = prev
            prev = current
            current = next_node
        return prev


# @leetup=code
